### PR TITLE
Yungong - displaying announcement table on play page updated

### LINKED ISSUE #31: Announcements Issue F3: Frontend Make Announcements appear on Play Page

### PR DESCRIPTION
**Pull request by yungongWang**
## Overview

In this PR, I only created `pagedAnnouncementFixtures.js`, `PagedAnnouncementTable.js`, and `PagedANnouncementTable.test.js` as my new component to show the announcement on play page, and I modified `CommonsOverview.js`, `CommonsOverview.test.js`, `PlayPaged.js`, `PlayPaged.tst.js` and `AdminViewPlayPaed.test.js` to make it compatible for my new frontend feature. The change on announcement controller file is not made by me, but I need them to deal with the case when enddate time is null, or there will be error and conflict. I pulled `AnnouncementController.java` and `AnouncementController.test.js` from Yixuan's branch.

I also deleted all my other files containing `//Stryker` comment accidentally pulled from Yixuan's unfinished branch and restored all these files to the same as those in the main branch, so they will not produce error message.

## Screenshots (Optional)

image
image

## Validation (Optional)

1. Enter the test commons
2. You can see a table that displays announcement by decreasing start date order.
3. You can click the blue button to test for hide / show announcement table function.
4. Noticing that because this task is only about frontend, all the code about creating and editing announcements are not updated in this PR, so there is no way to create announcements. However, there are some announcements that I created before, which are shown on the table in play page. So, you can check them for their time and sorting.
5. dokku link: https://happycows-dev-yungong.dokku-12.cs.ucsb.edu/

image
image

## Notification
There is one survived case for Stryker mutation test. This is because I used PagedProfitsTable.js as an example. I called `React.useMemo` as filter for qualified start and end date, and when the same thing is used in PagedProfitsTable.js, it disabled the Stryker test for the returned value in `[ ]`. So, after talking with TA, I also put a comment before this line and wonder if I can ignore the Stryker test for this line.

image
image
image

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #31

### CODE REVIEW

File: frontend/src/main/components/Commons/PagedAnnouncementTable.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import { timestampToDate } from "main/utils/dateUtils";
+
+
+const PagedAnnouncementTable = () => {
+
+    const refreshJobsIntervalMilliseconds = 5000;
+
+    const { commonsId } = useParams();
+
+    // Stryker disable all
```
Comment (yungongWang) [inline frontend/src/main/components/Commons/PagedAnnouncementTable.js:14]: I use PagedProfitsTable.js as example, and in PagedProfitsTable.js, it has these two `//Stryker` comment, so I also leave them here, I do not know if I can do so, please let me know if I need to write test

File: frontend/src/main/components/Commons/CommonsOverview.js [hunk 2]
```diff
@@ -3,11 +3,14 @@ import { Row, Card, Col, Button } from "react-bootstrap";
 import { useNavigate } from "react-router-dom";
 import { hasRole } from "main/utils/currentUser";
 import { daysSinceTimestamp } from "main/utils/dateUtils";
+import PagedAnnouncementTable from "main/components/Commons/PagedAnnouncementTable";
 
 export default function CommonsOverview({ commonsPlus, currentUser }) {
 
     let navigate = useNavigate();
     // Stryker disable next-line all
+    const [isTableVisible, setIsTableVisible] = React.useState(true);
```
Comment (yungongWang) [inline frontend/src/main/components/Commons/CommonsOverview.js:12]: The `Stryker` comments is there originally, and I just add another code after it without noticing it. Please let me know if I need to write test for these two line

**Review by jzipkin1812 — DISMISSED** [review]
I'm going to trust you that the two emphasized "//Stryker disable" comments are standard/inherited from previous examples of similar code from the codebase. I do see that the PagedProfitsTable has very similar stryker disable comments as yours so I think that's OK. Additionally, everything on the deployment works as expected. I have no complaints but staff may want you to add additional tests since I don't exactly know their standard on mutation tests.

**Review by pconrad — APPROVED** [review]
Code LGTM.. will test on dokku

**Pull request merged**